### ISSUE TITLE
When is the miner windows going to be brought back?  AMD gpu's are crippled til then

### ISSUE THREAD

**Issue by alucard20724** [MultiPoolMiner/MultiPoolMiner#1529]
there are several aglo out there for AMD cards that can't be used because mpm can't support them in the current version, such as x16r, x16s, xevan, skunk, lyra2z, lyra2re2 all of which have optimized sgminers for them, but they are all disabled in the mpm version that prevents miner window pop ups.

now mpm is just a miner for nvidia and claymore.

so when will mpm miner release a version that supports miner window opening?   The current version of mpm is a crutch/handicap for AMD cards.

it's already two weeks past from when it was suppose to be released and there isn't any detailed update.

**Comment by grantemsley** [comment]
Aaron's latest commit fixes sgminer.  Seems to be working well so far.

**Comment by alucard20724** [comment]
[SgminerX16r-4028_2018-04-02_22-18-01.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1870079/SgminerX16r-4028_2018-04-02_22-18-01.txt)

and it's not working.

**Comment by grantemsley** [comment]
I don't think sgminer x16r is in the master branch yet... Can you post the file from the miners directory you are using?

Are the other sgminers working now?

**Comment by alucard20724** [comment]
it's just like the other optimized sgminer files. of which none of them work because of the "redirection not supported" issues.  

using module ..\Include.psm1

$Path = ".\Bin\AMD_X16\sgminer.exe"
$Uri = "https://brian112358/sgminer-x16r/releases/download/v0.4.0/sgminer-x16r-v0.4.0-windows.zip"

$Commands = [PSCustomObject]@{

    "x16r" = " -I 19" #X16r
    "x16s" = " -I 19" #X16s

}

$Name = Get-Item $MyInvocation.MyCommand.Path | Select-Object -ExpandProperty BaseName

$Commands | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name | Where-Object {$Pools.(Get-Algorithm $_).Protocol -eq "stratum+tcp" <#temp fix#>} | ForEach-Object {
    [PSCustomObject]@{
        Type = "AMD"
        Path = $Path
        Arguments = "--api-listen -k $_ -o $($Pools.(Get-Algorithm $_).Protocol)://$($Pools.(Get-Algorithm $_).Host):$($Pools.(Get-Algorithm $_).Port) -u $($Pools.(Get-Algorithm $_).User) -p $($Pools.(Get-Algorithm $_).Pass)$($Commands.$_) --gpu-platform $([array]::IndexOf(([OpenCl.Platform]::GetPlatformIDs() | Select-Object -ExpandProperty Vendor), 'Advanced Micro Devices, Inc.'))"
        HashRates = [PSCustomObject]@{(Get-Algorithm $_) = $Stats."$($Name)_$(Get-Algorithm $_)_HashRate".Week}
        API = "Xgminer"
        Port = 4028
        URI = $Uri
    }
}

**Comment by grantemsley** [comment]
Yeah, it's missing the --text-only switch that was added to all of them in commit  0957a0aff710603f04e6b7b67d220da87db4987e

**Comment by alucard20724** [comment]
well.. the text starts but doesn't start/finish the benchmark

**Comment by alucard20724** [comment]
@grantemsley   I've just tested the --text-only switch.  It works for your supported optimized sgminer algos... just not x16r.  though i can get x16r working by itself.

**Comment by alucard20724** [comment]
@grantemsley upon further testing, it would appear that mpm is not shutting down sgminer when switching to a new algorithm.  With the windows hidden, you can't tell if sgminer is not shut down or not.  I check the task manager, and sure enough, sgminer was still running even though i had shut down mpm.

The reason i discovered this, was when I started megaminer, my hash rate was 10% of what it normally is megaminer (i use megaminer for my AMD, and Nvidia gpus now, still use mpm for cpu as these problems don't seem to affect cpu mining).  Task manager showed i had sgminer running, even though i had shut down all miners.

it would appear that the --text-only switch will get the mining started, but doesn't allow it to shut down.

**Comment by alucard20724** [comment]
@aaronsace the problem wasn't with megaminer, it was with mpm.  mpm isn't shutting down the optimized sgminers.  the --text-only switch that you implemented, allows the used of the optimized sgminers, but now the optimized sgminers are not being shut down when with mpm.

optimized sgminers wasn't working until you added the --text-only switch, which leads me to believe that the problem is still with mpm.

i don't see this problem at all with any of my systems that i run just megaminer on. (14 of them... various nvidia and amd)  I run megaminer on my systems now because it works for all the optimized sgminers for AMD, (except for cpu, i still use mpm for cpu... i run amd,nvidia, cpu from different folders).

I switched to megaminer last week because i got tired of this not being fixed.  Megaminer doesn't block window miner popup, which allows a hardware guy like me realtime feedback.

The problem lies somewhere in the codebase for hiding miner windows.
